### PR TITLE
mon: add mgr metdata commands, and overall 'versions' command for all daemon versions

### PR DESCRIPTION
**Pull request by liewegas**
<pre>
$ ceph versions
{
    "mon": {
        "ceph version 12.1.1-235-g9c3232d (9c3232d4c9375963f7f5f6c04b035788a13c0b79) luminous (rc)": 1
    },
    "mgr": {
        "ceph version 12.1.1-233-g1b51c63 (1b51c63d72ec10fd176af672e9cca7a85226021e) luminous (rc)": 2
    },
    "osd": {
        "ceph version 12.1.1-233-g1b51c63 (1b51c63d72ec10fd176af672e9cca7a85226021e) luminous (rc)": 1
    },
    "mds": {
        "ceph version 12.1.1-235-g9c3232d (9c3232d4c9375963f7f5f6c04b035788a13c0b79) luminous (rc)": 1
    },
    "rgw": {
        "ceph version 12.1.1-233-g1b51c63 (1b51c63d72ec10fd176af672e9cca7a85226021e) luminous (rc)": 1
    },
    "overall": {
        "ceph version 12.1.1-233-g1b51c63 (1b51c63d72ec10fd176af672e9cca7a85226021e) luminous (rc)": 4,
        "ceph version 12.1.1-235-g9c3232d (9c3232d4c9375963f7f5f6c04b035788a13c0b79) luminous (rc)": 2
    }
}
</pre>

### CODE REVIEW

File: doc/release-notes.rst [hunk 1]
```diff
@@ -158,10 +158,12 @@ Major Changes from Kraken
   * *CLI changes*:
 
     - The ``ceph -s`` or ``ceph status`` command has a fresh look.
-    - ``ceph {osd,mds,mon} versions`` summarizes versions of running daemons.
-    - ``ceph {osd,mds,mon} count-metadata <property>`` similarly
+    - ``ceph mgr metadata`` will dump metadata associated with each mgr
```
Comment (tchaikov) [inline doc/release-notes.rst]: add `FIXME`, if we plan to fix the man page later on.

File: src/mgr/ServiceMap.h [hunk 2]
```diff
@@ -47,10 +47,23 @@ struct ServiceMap {
...
       }
       std::ostringstream ss;
-      ss << daemons.size() << (daemons.size() > 1 ? "daemonss" : "daemon")
+      ss << daemons.size() << (daemons.size() > 1 ? " daemons" : " daemon")
 	 << " active";
       return ss.str();
     }
+
+    void count_metadata(const string& field,
+			std::map<std::string,int> *out) const {
```
Comment (tchaikov) [inline src/mgr/ServiceMap.h]: nit, s/int/unsigned/

File: src/mon/OSDMonitor.h [hunk 3]
```diff
@@ -428,6 +428,9 @@ class OSDMonitor : public PaxosService {
 
   int load_metadata(int osd, map<string, string>& m, ostream *err);
   void count_metadata(const string& field, Formatter *f);
+public:
+  void count_metadata(const string& field, map<string,int> *out);
+protected:
```
Comment (tchaikov) [inline src/mon/OSDMonitor.h]: nit, do we want to inherit from `OSDMonitor`?

**Comment by liewegas** [comment]
updated and rebased

File: src/mon/OSDMonitor.h (hunk 3 above)
Comment (liewegas) [inline src/mon/OSDMonitor.h]: I'm pretty unhappy about the duplication of hte metadata code across all teh *Monitor classes but factoring it out is a non-trivial cleanup due to a few inconsistencies.  Kicking that can down the road...

File: doc/release-notes.rst (hunk 1 above)
Comment (liewegas) [inline doc/release-notes.rst]: updated man page

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-07-21_21:27:50-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**